pkg/server/warcserver/daterange.go

```diff
@@ -18,24 +18,130 @@ package warcserver
 
 import (
 	"fmt"
+	"math"
+	"strings"
+	"time"
 )
 
 type DateRange struct {
-	from string
-	to   string
+	from int64
+	to   int64
+}
+
+const timeLayout = "20060102150405"
+
+func NewDateRange(fromstr string, tostr string) (DateRange, error) {
+	from, err := From(fromstr)
+	if err != nil {
+		return DateRange{0, 0}, err
+	}
+	to, err := To(tostr)
+	if err != nil {
+		return DateRange{0, 0}, err
+	}
+
+	return DateRange{
+		from,
+		to,
+	}, nil
 }
 
 // contains returns true if the timestamp ts contained by the bounds defined by the DateRange d.
-func (d DateRange) contains(ts string) bool {
-	return ts >= d.from && ts <= d.to
+// input 'ts' is 'trusted' and does not have the same parsing complexity as a From or To string
+func (d DateRange) contains(ts string) (bool, error) {
+	timestamp, err := time.Parse(timeLayout, ts)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse ts: %w", err)
+	}
+	unixTs := timestamp.Unix()
+
+	return unixTs >= d.from && unixTs <= d.to, nil
 }
 
-// From pads the timestamp f with 0's on the right until the string is 14 characters in length.
-func From(f string) string {
-	return fmt.Sprintf("%s%0*d", f, 14-len(f), 0)
+// Implemented according to https://pywb.readthedocs.io/en/latest/manual/cdxserver_api.html#from-to
+func From(f string) (int64, error) {
+	fLen := len(f)
+	if fLen%2 != 0 {
+		return 0, fmt.Errorf("'from' string was not an odd number, len: %d", fLen)
```

Comment:
Confusing and erroneous error message. If length of input is odd the timestamp cannot be parsed, but the error message claims the string was not odd.